src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala

```diff
@@ -3,26 +3,37 @@ package org.zalando.spearheads.innkeeper.oauth
 import com.google.inject.{Inject, Singleton}
 import org.slf4j.LoggerFactory
 import org.zalando.spearheads.innkeeper.oauth.OAuthJsonProtocol.authorizedUserFormat
+import org.zalando.spearheads.innkeeper.services.ServiceResult
+import org.zalando.spearheads.innkeeper.services.ServiceResult.{Ex, Result}
 import org.zalando.spearheads.innkeeper.utils.{OAuthServiceClient, EnvConfig, HttpClient}
-
-import scala.util.Try
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
 
 /**
  * @author dpersa
  */
 trait AuthService {
-  def authenticate(token: String): Try[AuthenticatedUser]
+  def authenticate(token: String): Future[Result[AuthenticatedUser]]
 }
 
 @Singleton
 class OAuthService @Inject() (
     config: EnvConfig,
-    @OAuthServiceClient() httpClient: HttpClient) extends AuthService {
+    @OAuthServiceClient() httpClient: HttpClient)(implicit val executionContext: ExecutionContext) extends AuthService {
 
   val logger = LoggerFactory.getLogger(this.getClass)
 
-  override def authenticate(token: String): Try[AuthenticatedUser] = {
-    httpClient.callJson(url(token), None).map(_.convertTo[AuthenticatedUser])
+  override def authenticate(token: String): Future[Result[AuthenticatedUser]] = {
+
+    httpClient.callJson(url(token), None).map { json =>
+      Try { json.convertTo[AuthenticatedUser] } match {
+        case Success(user) => ServiceResult.Success(user)
+        case Failure(ex) =>
+          logger.error(s"OAuthService unmarshalling failed with exception $ex")
```

Comment:
Ok, I see you commented on the same as me above :) I see, perhaps not needed then :)